crates/task-impls/src/quorum_proposal.rs

```diff
@@ -763,7 +644,14 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalTaskState<TYPE
             HotShotEvent::QuorumProposalSend(proposal, _) => {
                 let view = proposal.data.view_number;
                 if !self.update_latest_proposed_view(view).await {
-                    warn!("Failed to update latest proposed view");
+                    tracing::trace!("Failed to update latest proposed view");
+                    return;
+                }
+            }
+            HotShotEvent::QuorumProposalRecv(proposal, _) => {
+                let view = proposal.data.view_number;
+                if !self.update_latest_proposed_view(view).await {
```

Comment:
I'm not sure if we should call `update_latest_proposed_view` when receiving a proposal. If the proposal is invalid (since it may be from another node), wouldn't this cause an incorrect view update? I feel it's safer to only call `update_latest_proposed_view` if we are the proposal sender or after validating the proposal.

Reply:
Ah yeah that makes sense.